File: docs/snippets/python/tool-call-message_main.py
# Example: Tool/Function call message
# Simulated Python implementation

xml_output = """<?xml version="1.0" encoding="utf-8"?>
<tool>
  <functionCall name="get_weather" callId="call_001">
    <arguments>{"location": "San Francisco", "unit": "celsius"}</arguments>
  </functionCall>
</tool>"""

print(xml_output)

File: docs/snippets/csharp/tool-call-message_main.cs
// using Microsoft.Agents.Xml.Generated.Models;
// using Microsoft.Agents.Protocol.Xml;

// Create agent message with tool call
var message = new AgentMessage
{
    AgentId = "agent-456",
    MessageId = "msg-call-1",
    Contents = new List<AIContent>
    {
        new FunctionCallContent
        {
            CallId = "call_abc123",
            Name = "get_weather",
            Arguments = "{\"location\": \"Seattle\", \"unit\": \"fahrenheit\"}"
        }
    }
};

var serializer = new MessageSerializer();
var xmlOutput = serializer.Serialize(message);
Console.WriteLine(xmlOutput);